lisa/tests/scheduler/load_tracking.py

```diff
@@ -738,11 +739,24 @@ def _plot_behaviour(self, ax, df, title, task_duty_cycle_pct):
         ax.legend()
 
     def _test_behaviour(self, signal_name, error_margin_pct, allowed_error_pct):
+	# Note: This test-case is only valid if executed at capacity == 1024
+
         res = ResultBundle.from_bool(True)
         task = self.rtapp_profile[self.task_name]
         phase = task.phases[0]
         cpu = phase.cpus[0]
 
+        # Due to simplifications in the PELT simulator and general misalignment between
+        # the simulated PELT signal and the traced one the error margin has to be fairly
+	# generous. The error sources include:
+	#
+	# 1. Misaligned PELT period boundaries (error <= 22)
+	# 2. Inaccurate duty cycle of rt_app task which can't be compensated for in the
+        #    PELT simulator as it only speaks integer PELT periods.
+        # 3. Nearest sample can be up to 512us off (error <= 11).
+        #
+        # For a 50% 16ms period task that amounts to 5-7% error on TC2.
+
```

Comment:
Is 16ms the only period we care about? If the period was changed to 20 or 32 ms, will the error margin be different then?

Reply:
The error is still there for other periods. The various error sources scale differently with period so it is non-trivial to work out what the true margin should be. Most likely as complicated as writing a more accurate PELT simulator.

Reply:
OK. What I was trying to get to if we changed the test to use a different period other than 16ms, will we need to change the error margin too then?

Reply:
Yes, most likely.

Reply:
OK I was hoping we can explicitly mention this in the comment but I see you already pushed an updated version and not sure if it's worth a re-spin just for that. Up to you.

Reply:
I can make it explicit that the error margin is workload dependent if that helps. I have already specified the errors that have an upper bound.

Reply:
Done